core/src/rpc.rs

```diff
@@ -732,9 +732,12 @@ impl JsonRpcRequestProcessor {
                     .runtime_handle
                     .block_on(
                         bigtable_ledger_storage
-                            .get_confirmed_blocks(start_slot, (end_slot - start_slot) as usize),
+                            .get_confirmed_blocks(start_slot, (end_slot - start_slot) as usize + 1), // increment limit by 1 to ensure returned range is inclusive of both start_slot and end_slot
                     )
-                    .unwrap_or_else(|_| vec![]));
+                    .unwrap_or_else(|_| vec![])
+                    .into_iter()
+                    .filter(|&slot| slot <= end_slot)
+                    .collect());
```

Comment:
Could `.retain()` here?

Though it could be argued that the BT returning different results was a bug.

Reply:
Oh thx, `retain()` is definitely better!